# Anti-AH in Aquaman 2 of Change.org

In [1]:
cd "../../Change.org - Data"

D:\test\mapbox-android-sample-master\src\New folder\Change.org - Data


In [2]:
import os
import pandas as pd

# all petitions comments

In [3]:
data = [i for i in os.listdir() if i.split('.')[-1] == "csv"]

In [4]:
data

['all_petitions_comments_20211026-132241.csv',
 'all_petitions_comments_20211112-095033.csv',
 'all_petitions_comments_4_months.csv',
 'all_petitions_milestones_20211026-130816.csv',
 'all_petitions_milestones_20211112-093758.csv',
 'aquaman_petitions_milestones.csv',
 'aquman_petitions_comments.csv',
 'change_org_comments_petition_slug_hollywood-boycott-aquaman-2-amber-heard_at_20211005-210520_limit_1000_offset_0_delay_ms_100.csv',
 'change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500 (1).csv',
 'change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500 (2).csv',
 'change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500.csv',
 'change_org_petition_slug_united-states-supreme-court-put-amber-heard-behind-bars_at_20210821-084803_max_number_signatures_30 (1).csv']

## petitions comments

In [5]:
df_petitions_comments_1 = pd.read_csv("all_petitions_comments_20211026-132241.csv")
df_petitions_comments_2 = pd.read_csv("all_petitions_comments_20211112-095033.csv")

In [6]:
df_petitions_comments_1.head(1)

index         id                                            comment  \
0      0  810211260  Johnny Depp is Innocent, Amber Heard is a lier...   

   parent_id  children_count commentable_type  commentable_id  likes     role  \
0          0               0            Event        14986339      1  comment   

             created_at  ...  user.photo.sizes.large.size.width  \
0  2020-12-07T02:33:21Z  ...                              400.0   

   user.photo.sizes.large.size.height  \
0                               400.0   

                         user.photo.sizes.xlarge.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   

  user.photo.sizes.xlarge.processing user.photo.sizes.xlarge.size.width  \
0                              False                              800.0   

  user.photo.sizes.xlarge.size.height user.roles  \
0                               800.0         []   

                             commentable_entity.slug  \
0  warner-brothers-save-johnny-depp-as-gellert-gr...   

                                      petition__slug status_manual  
0  warner-brothers-save-johnny-depp-as-gellert-gr...       against  

[1 rows x 45 columns]

In [7]:
df_petitions_comments_1.shape

(17245, 45)

In [8]:
df_petitions_comments_2.head(1)

index         id                                            comment  \
0      0  810211260  Johnny Depp is Innocent, Amber Heard is a lier...   

   parent_id  children_count commentable_type  commentable_id  likes     role  \
0          0               0            Event        14986339      1  comment   

             created_at  ...  user.photo.sizes.large.size.width  \
0  2020-12-07T02:33:21Z  ...                              400.0   

   user.photo.sizes.large.size.height  \
0                               400.0   

                         user.photo.sizes.xlarge.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   

  user.photo.sizes.xlarge.processing user.photo.sizes.xlarge.size.width  \
0                              False                              800.0   

  user.photo.sizes.xlarge.size.height user.roles  \
0                               800.0         []   

                             commentable_entity.slug  \
0  warner-brothers-save-johnny-depp-as-gellert-gr...   

                                      petition__slug status_manual  
0  warner-brothers-save-johnny-depp-as-gellert-gr...       against  

[1 rows x 45 columns]

In [9]:
df_petitions_comments_2.shape

(27238, 45)

if we conactenated two dfs we shold get 27238 as the data we have scrapped twice
- at 2021-10-26
- 2021-11-26

In [10]:
df_petitions_comments = pd.concat([df_petitions_comments_1, df_petitions_comments_2]).drop_duplicates().drop(columns="index").reset_index(drop=True)

In [11]:
df_petitions_comments.shape

(27238, 44)

In [12]:
pd.options.display.max_columns = 50
df_petitions_comments.head(2)

id                                            comment  parent_id  \
0  810211260  Johnny Depp is Innocent, Amber Heard is a lier...          0   
1  809954316  She's an evil abuser. Justice for JD! She shou...          0   

   children_count commentable_type  commentable_id  likes     role  \
0               0            Event        14986339      1  comment   
1               0            Event        14986339      0  comment   

             created_at  deleted_at     user.id   user.slug user.first_name  \
0  2020-12-07T02:33:21Z         NaN   577958717   577958717           Cesar   
1  2020-12-01T19:45:00Z         NaN  1166768300  1166768300          Brandi   

  user.last_name user.short_display_name user.display_name      user.city  \
0        Ramirez                   Cesar     Cesar Ramirez      Santa Ana   
1        Granger                  Brandi    Brandi Granger  Missouri City   

  user.state_code user.country_code user.website user.locale user.description  \
0              CA                US          NaN       en-US              NaN   
1              TX                US          NaN       en-US              NaN   

   user.photo.id                               user.photo.url  \
0    145246649.0  photos/7/uq/dv/QVuqDvrhjPwmlbz-fullsize.jpg   
1            NaN                                          NaN   

                          user.photo.sizes.small.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.small.processing  user.photo.sizes.small.size.width  \
0                             False                               48.0   
1                               NaN                                NaN   

   user.photo.sizes.small.size.height  \
0                                48.0   
1                                 NaN   

                         user.photo.sizes.medium.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.medium.processing  user.photo.sizes.medium.size.width  \
0                              False                               128.0   
1                                NaN                                 NaN   

   user.photo.sizes.medium.size.height  \
0                                128.0   
1                                  NaN   

                          user.photo.sizes.large.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.large.processing  user.photo.sizes.large.size.width  \
0                             False                              400.0   
1                               NaN                                NaN   

   user.photo.sizes.large.size.height  \
0                               400.0   
1                                 NaN   

                         user.photo.sizes.xlarge.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.xlarge.processing  user.photo.sizes.xlarge.size.width  \
0                              False                               800.0   
1                                NaN                                 NaN   

   user.photo.sizes.xlarge.size.height user.roles  \
0                                800.0         []   
1                                  NaN         []   

                             commentable_entity.slug  \
0  warner-brothers-save-johnny-depp-as-gellert-gr...   
1  warner-brothers-save-johnny-depp-as-gellert-gr...   

                                      petition__slug status_manual  
0  warner-brothers-save-johnny-depp-as-gellert-gr...       against  
1  warner-brothers-save-johnny-depp-as-gellert-gr...       against

In [13]:
df_petitions_comments[df_petitions_comments["commentable_entity.slug"].str.contains('dc-entertainment-remove-amber-heard-from-aquaman-2')]["commentable_entity.slug"].unique()

array(['dc-entertainment-remove-amber-heard-from-aquaman-2'], dtype=object)

In [14]:
aquman_petitions_comments = df_petitions_comments[df_petitions_comments["commentable_entity.slug"].str.contains('dc-entertainment-remove-amber-heard-from-aquaman-2')]

In [15]:
aquman_petitions_comments = aquman_petitions_comments.reset_index(drop=True)

In [18]:
aquman_petitions_comments

id                                            comment  parent_id  \
0     825956692                                         Fuck Amber          0   
1     825930086  Jonathan Blake Owens this is despicable from H...          0   
2     825868765  Потому что она должна ответить за ложь и недос...          0   
3     825850623                     All my niggas hate Amber Heard          0   
4     825844869  Amber Heard is a nasty liar who doesn't deserv...          0   
...         ...                                                ...        ...   
4995  809327580  I’ve been domestic beaten and bullied by a fem...          0   
4996  809327534                        I am against domestic abuse          0   
4997  809327530                                        Desgraciada          0   
4998  809327520       Weil diese frau es nicht anders verdient hat          0   
4999  809327505                                I love Jack sparrow          0   

      children_count commentable_type  commentable_id  likes     role  \
0                  0            Event        14451136      0  comment   
1                  0            Event        14451136      0  comment   
2                  0            Event        14451136      0  comment   
3                  0            Event        14451136      0  comment   
4                  0            Event        14451136      0  comment   
...              ...              ...             ...    ...      ...   
4995               0            Event        14451136      2  comment   
4996               0            Event        14451136      0  comment   
4997               0            Event        14451136      0  comment   
4998               0            Event        14451136      0  comment   
4999               0            Event        14451136      0  comment   

                created_at  deleted_at     user.id   user.slug  \
0     2021-11-12T12:52:49Z         NaN   610108946   610108946   
1     2021-11-11T14:39:27Z         NaN  1192010378  1192010378   
2     2021-11-09T19:32:21Z         NaN  1233208172  1233208172   
3     2021-11-09T10:35:36Z         NaN  1188639876  1188639876   
4     2021-11-09T06:26:34Z         NaN   316491339   316491339   
...                    ...         ...         ...         ...   
4995  2020-11-16T12:14:05Z         NaN   436788126   436788126   
4996  2020-11-16T12:11:42Z         NaN  1126767494  1126767494   
4997  2020-11-16T12:11:30Z         NaN   516583541   516583541   
4998  2020-11-16T12:11:04Z         NaN  1164562735  1164562735   
4999  2020-11-16T12:09:55Z         NaN   975707501   975707501   

     user.first_name user.last_name user.short_display_name  \
0              Robin    van Rooijen                   Robin   
1           Jonathan          Owens                Jonathan   
2            Vitaliy       Shevchuk                 Vitaliy   
3               Musa          Keita                    Musa   
4            Cynthia              B                 Cynthia   
...              ...            ...                     ...   
4995          Marcus        Ambrose                  Marcus   
4996      Daijinlung        Thaimei              Daijinlung   
4997       Scarlette        Arriaza               Scarlette   
4998            Dean          Rubin                    Dean   
4999         Jaydeep        Ganguly                 Jaydeep   

       user.display_name    user.city user.state_code user.country_code  \
0      Robin van Rooijen          NaN             NaN                NL   
1         Jonathan Owens  Westminster              SC                US   
2       Vitaliy Shevchuk          NaN             NaN                UA   
3             Musa Keita        Accra             NaN                GH   
4              Cynthia B       Oshawa             NaN                CA   
...                  ...          ...             ...               ...   
4995      Marcus Ambrose  Northampton             ENG                GB   
4996  Daijinl

In [19]:
aquman_petitions_comments.to_csv("aquman_petitions_comments.csv", index=False)

In [20]:
pd.read_csv(
    "change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500.csv"
)

_index      _type                    _id    _score  \
0    petitions-en  petitions  20569720::Amber Heard  7.115500   
1    petitions-en  petitions  26062476::Amber Heard  6.871536   
2    petitions-en  petitions  25704328::Amber Heard  6.744285   
3    petitions-en  petitions  25752458::Amber Heard  6.438685   
4    petitions-en  petitions  13641580::Amber Heard  6.325689   
..            ...        ...                    ...       ...   
288  petitions-en  petitions   4373742::Amber Heard  0.669221   
289  petitions-en  petitions  23141324::Amber Heard  0.633609   
290  petitions-en  petitions  13269107::Amber Heard  0.567439   
291  petitions-en  petitions               13392951  0.507148   
292  petitions-en  petitions               14168553  0.453598   

     _source__total_signature_count _source__created_locale  \
0                             23844                   en-US   
1                             13594                   en-US   
2                             34486                  es-419   
3                             38987                   en-US   
4                             11460                   ru-RU   
..                              ...                     ...   
288                               5                   en-US   
289                              12                   en-US   
290                               6                   en-GB   
291                               8                   en-US   
292                               7                  es-419   

    _source__first_name  _source__weekly_signature_count  \
0                 Linda                              324   
1              Kasandra                               77   
2                Alexis                              116   
3               Dominic                               31   
4                   Ksu                                5   
..                  ...                              ...   
288           Michael S                                0   
289              Hunter                                0   
290                 Tom                                0   
291             Crystal                                0   
292               Bruce                                1   

    _source__recent_activity_date  _source__is_pledge  \
0            2020-03-01T07:41:24Z               False   
1            2021-01-14T22:55:30Z               False   
2            2020-11-07T02:37:54Z               False   
3            2021-05-15T19:11:39Z               False   
4            2019-11-16T18:31:44Z               False   
..                            ...                 ...   
288          2015-09-28T21:04:19Z               False   
289          2020-06-28T21:44:07Z               False   
290          2018-06-09T01:51:59Z               False   
291          2018-07-19T22:06:26Z               False   
292          2019-01-12T14:41:48Z               False   

                        _source__targeting_description   _source__created_at  \
0                                            Hollywood  2020-03-01T07:41:24Z   
1           United States Supreme Court, UK Parliament  2020-11-28T03:46:49Z   
2    PEople with common sense, Johnny Depp, Warner ...  2020-11-07T01:14:06Z   
3           dc, Warner Bros, Amber Heard, danny devito  2020-11-09T15:57:43Z   
4                                          Johnny Depp  2018-10-05T12:57:38Z   
..                                                 ...                   ...   
288                                         Tim Butler  2015-09-28T21:04:19Z   
289  U​.​S. House of Representatives, U​.​S. Senate...  2020-06-28T21:44:07Z   
290                                     Boag's Brewery  2018-06-09T01:51:59Z   
291                                  darnajacpulch1986  2018-07-19T22:06:26Z   
292                                             Rainey  2019-01-12T14:41:48Z   

     _source__user_id  _source__victory _source__victory_date  \
0          1046163930             False  

In [21]:
df_all_petitions_milestones = pd.read_csv(
    "all_petitions_milestones_20211112-093758.csv"
)

In [24]:
df_all_petitions_milestones[df_all_petitions_milestones.petition__slug.str.contains("dc-entertainment-remove-amber-heard-from-aquaman-2")].petition__slug.unique()

array(['dc-entertainment-remove-amber-heard-from-aquaman-2'], dtype=object)

In [25]:
aquaman_petitions_milestones = df_all_petitions_milestones[df_all_petitions_milestones.petition__slug.str.contains("dc-entertainment-remove-amber-heard-from-aquaman-2")].reset_index(drop=True).drop(columns="index")

In [26]:
aquaman_petitions_milestones

__typename        id                 createdAt  \
0      StarterPetitionUpdate  29715353  2021-10-13T04:00:09.000Z   
1      StarterPetitionUpdate  29694389  2021-10-07T13:28:29.000Z   
2      StarterPetitionUpdate  29671480  2021-10-01T04:03:38.000Z   
3      StarterPetitionUpdate  29650692  2021-09-26T00:38:23.000Z   
4      StarterPetitionUpdate  29579670  2021-09-09T14:53:24.000Z   
..                       ...       ...                       ...   
301  MilestonePetitionUpdate  24317632  2019-03-20T13:29:22.000Z   
302  MilestonePetitionUpdate  24317155  2019-03-20T11:12:40.000Z   
303  MilestonePetitionUpdate  24316808  2019-03-20T08:49:06.000Z   
304  MilestonePetitionUpdate  24316460  2019-03-20T06:18:51.000Z   
305  MilestonePetitionUpdate  24316273  2019-03-20T04:55:31.000Z   

     signatureCount                                     petition__slug  \
0               NaN  dc-entertainment-remove-amber-heard-from-aquam...   
1               NaN  dc-entertainment-remove-amber-heard-from-aquam...   
2               NaN  dc-entertainment-remove-amber-heard-from-aquam...   
3               NaN  dc-entertainment-remove-amber-heard-from-aquam...   
4               NaN  dc-entertainment-remove-amber-heard-from-aquam...   
..              ...                                                ...   
301           100.0  dc-entertainment-remove-amber-heard-from-aquam...   
302            50.0  dc-entertainment-remove-amber-heard-from-aquam...   
303            25.0  dc-entertainment-remove-amber-heard-from-aquam...   
304            10.0  dc-entertainment-remove-amber-heard-from-aquam...   
305             5.0  dc-entertainment-remove-amber-heard-from-aquam...   

                                         title  \
0                               Appeal denied!   
1                Amber Heard hits Jason Momoa?   
2               Petition to charge Amber Heard   
3      Why Amber Heard lied and accused Johnny   
4    Mental Health Evaluation for Amber Heard?   
..                                         ...   
301                                        NaN   
302                                        NaN   
303                                        NaN   
304                                        NaN   
305                                        NaN   

                                           description  \
0    <p>Amber Heard has tried at least three times ...   
1    <p>Filming is suspended on Aquaman 2, accordin...   
2    <p>There's a new petition making the rounds, t...   
3    <p>Throughout the time that this petition has ...   
4    <p>Team Depp is seeking to compel Ms. Heard to...   
..                                                 ...   
301                                                NaN   
302                                                NaN   
303                                                NaN   
304                                                NaN   
305                                                NaN   

                                                 photo video embeddedMedia  \
0                                                  NaN   NaN           NaN   
1    {'id': '225992326', 'petitionMedium': {'url': ...   NaN           NaN   
2    {'id': '225977590', 'petitionMedium': {'url': ...   NaN           NaN   
3    {'id': '225964688', 'petitionMedium': {'url': ...   NaN           NaN   
4    {'id': '225920921', 'petitionMedium': {'url': ...   NaN           NaN   
..                                                 ...   ...           ...   
301                                                NaN   NaN           NaN   
302                                                NaN   NaN           NaN   
303                                                NaN   NaN           NaN   
304                                                NaN   NaN           NaN   
305                                                NaN   NaN           NaN   

     dmResponse status_manual  
0           NaN       against  
1           N

In [27]:
aquaman_petitions_milestones.to_csv("aquaman_petitions_milestones.csv", index=False)